internal/compiler/widgets/native/std-widgets.slint

```diff
@@ -257,3 +257,88 @@ export GridBox := GridLayout {
     spacing: NativeStyleMetrics.layout-spacing;
     padding: NativeStyleMetrics.layout-spacing;
 }
+
+export StandardTableView := Rectangle {
+    callback sort-ascending(int);
+    callback sort-descending(int);
+    
+    out property <int> current-sort-column: -1;
+    in-out property <[TableColumn]> columns;
+    in property <[[StandardListViewItem]]> rows;
+    in property <length> min-header-height: 42px;
+   
+    min-width: 400px;
+    min-height: 200px;
+    horizontal-stretch: 1;
+    vertical-stretch: 1;
+
+    function sort(index: int) {
+        if (current-sort-column != index) {
+            columns[current-sort-column].sort-by-ascending = false;
+            columns[current-sort-column].sort-by-descending = false;
+        }
+
+        if(columns[index].sort-by-ascending) {
+            columns[index].sort-by-ascending = false;
+            columns[index].sort-by-descending = true;
+            sort-descending(index);
+        } else {
+            columns[index].sort-by-ascending = true;
+            columns[index].sort-by-descending = false;
+            sort-ascending(index);
+        }
+
+        current-sort-column = index;
+    }
+
+    VerticalLayout {  
+        HorizontalLayout {  
+            for column[index] in columns : NativeTableColumn {
+                item: column;
+                horizontal-stretch: column.horizontal-stretch;
+
+                TouchArea {  
+                    clicked => {  
+                        sort(index);
+                    }
+                }
+            }
+        }
+
+        scroll-view := ScrollView {  
+            vertical-stretch: 1;
+
+            VerticalLayout {  
+                alignment: start;
+                
+                for row[i] in rows : HorizontalLayout {   
```

Comment:
The rows also should be drawn by the style.  (can be done in the followup PR) )

Reply:
Added to #2033